### PR TITLE
Add link to Chisel Breakdown slides to README

### PR DESCRIPTION
**Pull request by jackkoenig**
Provide some useful developer documentation

### Contributor Checklist

- [NA] Did you add Scaladoc to every public function/method?
- [NA] Did you add at least one test demonstrating the PR?
- [NA] Did you delete any extraneous printlns/debugging code?
- [x] Did you specify the type of improvement?
- [NA] Did you add appropriate documentation in `docs/src`?
- [x] Did you state the API impact?
- [x] Did you specify the code generation impact?
- [x] Did you request a desired merge strategy?
- [NA] Did you add text to be included in the Release Notes for this change?

#### Type of Improvement

  - documentation         

#### API Impact

No impact

#### Backend Code Generation Impact

No impact

#### Desired Merge Strategy

 - Squash

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you add the appropriate labels?
- [ ] Did you mark the proper milestone (3.2.x, 3.3.x, 3.4.x, 3.5.0) ?
- [ ] Did you review?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you mark as `Please Merge`?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -151,6 +151,12 @@ These are the base data types for defining circuit components:
 ## Contributor Documentation
 This section describes how to get started contributing to Chisel itself, including how to test your version locally against other projects that pull in Chisel using [sbt's managed dependencies](https://www.scala-sbt.org/1.x/docs/Library-Dependencies.html).
 
+### Useful Resources for Contributors
+
+The [Useful Resources](#useful-resources) for users are also helpful for contributors
```
Comment (seldridge) [inline README.md:156]:
```suggestion
The [Useful Resources](#useful-resources) for users are also helpful for contributors.
```

**Review by seldridge — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -151,6 +151,12 @@ These are the base data types for defining circuit components:
...
+
+- [**Chisel Breakdown Slides**](https://docs.google.com/presentation/d/114YihixFBPCfUnv1inqAL8UjsiWfcNWdPHX7SeqlRQc/edit?usp=sharing), an introductory talk about Chisel's internals
```
[the remark this answers is not in the document]
Comment (jackkoenig) [reply README.md:158]: I had just copy-pasted this from the Google sharing link, but it looks like we can do away with everything `/edit` on so I'll just do that.

Comment (jackkoenig) [inline README.md:158]:
```suggestion
- [**Chisel Breakdown Slides**](https://docs.google.com/presentation/d/114YihixFBPCfUnv1inqAL8UjsiWfcNWdPHX7SeqlRQc), an introductory talk about Chisel's internals
```

[the remark this answers is not in the document]
Comment (sequencer) [reply README.md]: we can even remove the `/edit`

[the remark this answers is not in the document]
Comment (jackkoenig) [reply README.md]: Yep! Did!

**Pull request merged**